_source/_posts/2020-10-22-set-up-a-mute-indicator-light-for-zoom-with-hammerspoon.md

```diff
@@ -0,0 +1,283 @@
+parser.add_argument("-d", "--delay", default=50, type=int, help="Light shift time in milliseconds")
+args = parser.parse_args()
+
+b1 = Blink1()
+b1.fade_to_rgb(args.delay, args.red, args.green, args.blue)
+```
+
+Copy the code above into a file named `blink.py` and then try running it from the command line:
+
+```shell
+python blink.py --green 128
+```
+
+This should turn the light green at about half brightness.
+
+Turn it off with this command:
+
+```shell
+python blink.py
+```
+
+
+## Connecting Hammerspoon to the blink(1) USB LED
+
+Now that you have a working Hammerspoon setup and a working Python script to control the blink(1) light, it's time to tie them together!
+
+Start by selecting **Open Config** from the Hammerspoon menu icon, then edit the `~/.hammerspoon/init.lua` file to look like the code below. Make sure to replace the path to the Python scripts with the path for your system!
+
+```lua
+updateZoomStatus = function(event)
+  hs.printf("updateZoomStatus(%s)", event)
+  if (event == "from-running-to-meeting") then
+    zoomStatusMenuBarItem:returnToMenuBar()
+  elseif (event == "muted") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py" "--red" "128"]])
+    zoomStatusMenuBarItem:setTitle("🔴")
+  elseif (event == "unmuted") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py" "--green" "128"]])
+    zoomStatusMenuBarItem:setTitle("🟢")
+  elseif (event == "from-meeting-to-running") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py"]])
+    zoomStatusMenuBarItem:removeFromMenuBar()
+  end
+end
+```
+
+Once you've finished modifying the `updateZoomStatus` function, save the file, then use the Hammerspoon menu icon to select the "Reload Config" menu item.
+
+Try it all out by reloading Zoom and starting a meeting. If it's all set up correctly, the blink(1) light should turn on shortly after the meeting starts!
+
+## Note for Zoom employees
+
+If you are an employee of Zoom, I have a special message for you: I wish I could have made a mute indicator light without Hammerspoon.
+
+My ideal way to integrate with Zoom would be to configure it to call a command line application every time the state changed inside of Zoom. In this dream scenario, I imagine opening up the Zoom preferences and putting the path to an executable file into a field somewhere. From then on, Zoom would call that executable file passing the event over the command line. In essence, I'd like to turn the `updateZoomStatus` Lua function in my Hammerspoon configuration into a command line script.
+
+## Did it work for you?
+
+As always, I'd love to hear what you thought of this post. Did it work for you? Did you run into issues? Let me know by commenting below or by reaching out to me on Twitter where I'm [@jf](https://twitter.com/jf).
```

Comment:
No links to other blog posts seems like a missed opportunity?. Maybe a Python post or two?